test/hotspot/jtreg/compiler/print/CompileCommandMemLimit.java

```diff
@@ -109,31 +110,47 @@ private static void test(String include, String exclude) throws Exception {
 
         oa.reportDiagnosticSummary();
 
-        String expectedNameIncl = getTestMethod(include)
-                .replace('.', '/')
-                .replace("$", "\\$");
-        String expectedNameExcl = getTestMethod(exclude)
-                .replace('.', '/')
-                .replace("$", "\\$");
-
+        String method1regex = testMethodNameForRegex(getTestMethod(METHOD1));
+        String method2regex = testMethodNameForRegex(getTestMethod(METHOD2));
+        String method3regex = testMethodNameForRegex(getTestMethod(METHOD3));
         String ct = c2 ? "c2" : "c1";
 
         if (test_crash) {
             oa.shouldNotHaveExitValue(0);
             oa.shouldMatch("# *Internal Error.*");
-            oa.shouldMatch("# *fatal error: " + ct + " *" + expectedNameIncl + ".*: Hit MemLimit .*limit: 4096.*");
-            oa.shouldNotMatch(".*" + expectedNameExcl + ".*");
+            oa.shouldMatch("# *fatal error: " + ct + " *" + method2regex + ".*: Hit MemLimit .*limit: 4096.*");
         } else {
-            // Should see trace output when methods are compiled
-            oa.shouldHaveExitValue(0)
-                    .shouldMatch(".*" + expectedNameIncl + ".*")
-                    .shouldNotMatch(".*" + expectedNameExcl + ".*");
+            oa.shouldHaveExitValue(0);
+
+            // In debug builds we have an inbuilt MemLimit. It is very high, so we don't expect it to fire in this test.
+            // But it will still show up in the final report.
+            String implicitMemoryLimit = Platform.isDebugBuild() ? "1024M" : "-";
+
+            // With C2, we print number of nodes, with C1 we don't
+            String numberNodesRegex = c2 ? "\\d+" : "-";
+
+            // method 2 should have hit its tiny limit
+            oa.shouldMatch(ct + " " + method2regex + ".*: Hit MemLimit \\(limit: 4096 now: \\d+\\)");
 
-            // Expect this log line
-            oa.shouldMatch(".*" + expectedNameIncl + ".*Hit MemLimit.*");
+            // neither of the other ones should have hit a limit
+            oa.shouldNotMatch(method1regex + ".*Hit MemLimit");
+            oa.shouldNotMatch(method3regex + ".*Hit MemLimit");
 
-            // Expect final output to contain "oom"
-            oa.shouldMatch(".*oom.*" + expectedNameIncl + ".*");
+            // Final report:
+            // Method 1 should show up as "ok" and with the default limit, e.g.
+            // total     NA        RA        result  #nodes  limit   time    type  #rc thread              method
+            // 32728     0         32728     ok     -       1024M   0.045   c1    1   0x000000011b019c10  compiler/print/CompileCommandMemLimit$TestMain::method1(()J)
+            oa.shouldMatch("\\d+ +\\d+ +\\d+ +ok +" + numberNodesRegex + " +" + implicitMemoryLimit + " +.* +" + method1regex);
```

Comment:
A minor suggestion regarding the regex. I find "\s+" more readable than " +" to match multiple spaces.